src/class-wpml-gutenberg-integration.php

```diff
@@ -196,6 +197,10 @@ private function render_inner_HTML( $block ) {
 			$content = $block->innerHTML;
 		}
 
+		if ( self::TOOLSET_FIELDS_AND_TEXT_BLOCK_NAME === $block->blockName ) {
```

Comment:
@kagg-design Do you know why we need this only for this block and not for other blocks? eg. What is between this and the wp/paragraph block?
Is there a way to make this more general so it can be used for other blocks if they need it also?

Reply:
@brucepearson I have tested other core/paragraph block and it is properly prepared (has `p` and `br` at this point).

Reply:
As it seems this behaviour is specific to Toolset, or at least, not the standard behaviour in Gutenberg, I suggest we allow third-party plugins to control that behaviour.

# Option 1

Replace `self::TOOLSET_FIELDS_AND_TEXT_BLOCK_NAME === $block->blockName` with `apply_filters( 'wpml_gutenberg_inner_HTML_content_autop,  false, $block, $content )`.

# Option 2

Provide a more generic filter such as `apply_filters( 'wpml_gutenberg_inner_HTML_content,  $content, $block )`.

# Option 3

Same as Option 2, but add the `wpml_inner_HTML_content` filter to the main **wpml-page-builders** project and include a third argument which specifies the type of integration (Gutenberg, Elementor, Cornerstone, etc.).

Reply:
@andreasciamanna Unfortunately, no filters can be used here. Toolset Views simply do not init class `\ToolsetBlocks\Block\Style\Block\FieldsAndText`. This class is the only logical point to add a filter. 

I have added filter to `wp-content/plugins/views/vendor/toolset/blocks/server/Block/Style/Block/Factory.php` and it works, but looks like a hack...

Reply:
More brave approach is to make `wpautop` without filter, like we do in `\WPML_Gutenberg_Integration::render_block` for Classic Block.

With those blocks I tested (for instance, simple paragraph), nothing wrong happens. Block inner html comes as `\n<p>text1<br />\ntext2<p>\n` and it is OK. After `wpautop` only first `\n` is missed, so it becomes `<p>text1<br />\ntext2<p>\n`.

Reply:
@kagg-design Mine was meant to be food for thought: let third-party plugins hook into our code.

Then I suggested three possible options.

I wasn't really suggesting to do any change at this stage since the plan was to let Toolset fix the problem for now: did we do that?

Reply:
The best approach for this is to add a format attribute to the config file and then handle in our code according to that. **format="wpautop"**

```
<gutenberg-block type="toolset-blocks/fields-and-text" translate="1">
	<xpath format="wpautop">//div[@class='tb-fields-and-text']</xpath>
</gutenberg-block>
```